camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusHourlySweeper.java

```diff
@@ -0,0 +1,327 @@
+
+  private final CamusSweeperMetrics metrics;
+
+  public CamusHourlySweeper() {
+    super();
+    metrics = new CamusSweeperMetrics();
+  }
+
+  public CamusHourlySweeper(Properties props) {
+    super(props);
+    metrics = new CamusSweeperMetrics();
+  }
+
+  @Override
+  public Map<FileStatus, String> findAllTopics(Path input, PathFilter filter, String topicSubdir, FileSystem fs)
+      throws IOException {
+    Map<FileStatus, String> topics = new HashMap<FileStatus, String>();
+    for (FileStatus f : fs.listStatus(input)) {
+      if (f.isDir() && filter.accept(f.getPath())) {
+        LOG.info("found topic: " + f.getPath().getName());
+        topics.put(fs.getFileStatus(f.getPath()), f.getPath().getName());
+      }
+    }
+    return topics;
+  }
+
+  @Override
+  public void run() throws Exception {
+    metrics.timeStart = System.currentTimeMillis();
+    super.run();
+    reportMetrics();
+    addOutliers();
+  }
+
+  private void addOutliers() throws IOException {
+    Configuration conf = new Configuration();
+    for (Entry<Object, Object> pair : props.entrySet()) {
+      String key = (String) pair.getKey();
+      conf.set(key, (String) pair.getValue());
+    }
+    for (Properties jobProps : planner.getOutlierProperties()) {
+      this.executorService.submit(new OutlierCollectorRunner(jobProps, FileSystem.get(conf)));
+    }
+  }
+
+  private void reportMetrics() throws IllegalArgumentException, IOException {
+    LOG.info("reporting metrics");
+    metrics.reportTotalRunningTime();
+    metrics.reportTotalDataSize();
+    metrics.reportDataSizeByTopic();
+    metrics.reportDurationFromStartToRunnerStart();
+    metrics.reportDurationFromRunnerStartToMRSubmitted();
+    metrics.reportDurationFromMRSubmittedToMRStarted();
+    metrics.reportDurationFromMRStartedToMRFinished();
+    LOG.info("finished reporting metrics");
+  }
+
+  public static void main(String args[]) throws Exception {
+    CamusSweeper job = new CamusHourlySweeper();
+    ToolRunner.run(job, args);
+  }
+
+  @Override
+  protected Future<?> runCollector(Properties props, String topic) {
+    String jobName = topic + "-" + UUID.randomUUID().toString();
+    props.put("tmp.path", props.getProperty(CAMUS_SWEEPER_TMP_DIR) + "/" + jobName + "_" + System.currentTimeMillis());
+
+    if (props.containsKey(REDUCE_COUNT_OVERRIDE + topic))
+      props.put("reducer.count", Integer.parseInt(props.getProperty(REDUCE_COUNT_OVERRIDE + topic)));
+
+    LOG.info("Processing " + props.get(INPUT_PATHS));
+
+    return executorService.submit(new KafkaCollectorRunner(jobName, props, errorMessages, topic));
+  }
+
+  @Override
+  protected void runCollectorForTopicDir(FileSystem fs, String topic, Path topicSourceDir, Path topicDestDir)
+      throws Exception {
+    LOG.info("Running collector for topic " + topic + " source:" + topicSourceDir + " dest:" + topicDestDir);
+    List<Future<?>> tasksToComplete = new ArrayList<Future<?>>();
```

Comment:
What is the use of this variable? tasksToComplete is not used anywhere after line 131